### PR TITLE
started using the open fda api

### PR DESCRIPTION
**Pull request by codingwithcn**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -10,7 +10,8 @@ import { type DrugProp } from '@/Drug.type';
 const resultRef = ref<{ result: DrugProp[] }>({ result: [] });
 const searchText = ref("");
 
-const searchDrugs = (drugSearch: string) => {
+const searchDrugs = async (drugSearch: string) => {
```
Comment (rbannal86) [inline src/App.vue:11]: While using the app, I am running into some very jagged behavior- When filling out the search bar, the app is freezing, especially when removing what I've typed into the input.

I appreciate the thought behind designing this app the way you have, but the search triggering a fetch each time the input updates isn't working the way we want it. I think scaling back to only searching when the submit button is clicked would improve behavior a bunch.

File: src/App.vue [hunk 2]
```diff
@@ -10,7 +10,8 @@ import { type DrugProp } from '@/Drug.type';
...
+  if ( drugSearch.length < 1 ) {
```
Comment (rbannal86) [inline src/App.vue]: I would like all of the static data removed from this app now that we're doing a lookup with the API.

File: src/App.vue [hunk 3]
```diff
@@ -21,7 +22,16 @@ const searchDrugs = (drugSearch: string) => {
             })()
           )
         : (() => drug.openfda !== undefined && drug.openfda.brand_name !== undefined)()
-    ) as DrugProp[]
+    ) as DrugProp[];
```
Comment (rbannal86) [inline src/App.vue]: The above code is difficult to understand at a quick glance. Ternaries are great, I love them, but they should generally be limited to single line returns and not multi-line logic. Also, always document the code if you start writing something that might be complicated.

File: src/App.vue (hunk 1 above)
Comment (codingwithcn) [reply src/App.vue:11]: [096002b](https://github.com/codingwithcn/OpenFDA-Vue/commit/096002b09e7f26f23adda863494bda79cdb10adc) fixed issue

File: src/App.vue (hunk 3 above)
Comment (codingwithcn) [reply src/App.vue]: [096002b](https://github.com/codingwithcn/OpenFDA-Vue/commit/096002b09e7f26f23adda863494bda79cdb10adc) fixed issue

File: src/App.vue (hunk 2 above)
Comment (codingwithcn) [reply src/App.vue]: [096002b](https://github.com/codingwithcn/OpenFDA-Vue/commit/096002b09e7f26f23adda863494bda79cdb10adc) fixed issue

**Comment by rbannal86** [comment]
For this project, I think we're good to merge this into the main branch. If this were code being reviewed for some production application, we would want to look into the following:

1. Restructure the app so the lookup and info displays are in a separate component so that the bare minimum of business logic is in App.vue.
2. Look into debounce packages so we can re-implement the `watchEffect` setup.
3. Clean up CSS (though, in a real project, we would probably want to use a component/UI library instead of doing everything custom).
4. Fix console warnings. There's something going on with a key somewhere.
5. Fix the additional information display/conditionally render them if there isn't any data that fits.

Overall, though, good job. If you want to talk more about any of the above points, let me know!

**Pull request merged**